bcipy/tasks/main_frame.py

```diff
@@ -228,15 +238,39 @@ def decide_state_update(self):
         self.list_epoch[-1]['decision'] = decision
         return decision
 
-    def prepare_stimuli(self):
+    # TODO: think about where timing, color, is_txt, num_sti should go.
+    # e.g. is_txt is predefined with the alphabet directly
+    def _prepare_stimuli(self, timing=[1, 0.2], color=['red', 'white'],
+                         num_sti=1, is_txt=True):
         """ Given the alphabet, under a rule, prepares a stimuli for
             the next sequence
             Return:
                 stimuli(tuple[list[char],list[float],list[str]]): tuple of
                     stimuli information. [0]: letter, [1]: timing, [2]: color
                 """
-        stimuli = \
-            best_case_rsvp_seq_gen(self.alphabet, self.list_epoch[-1][
-                'list_distribution'][-1], num_sti=1, is_txt=self.is_txt_sti,
-                timing=self.stimuli_timing)
+
+        tmp = self.list_epoch[-1]['list_distribution'][-1][:]
```

Comment:
There are several places in the code that use this selector `self.list_epoch[-1]['list_distribution'][-1]`. It would clarify what this is doing if this was pulled out into another method with a more descriptive name (latest_prob_distribution?) and some documentation. Then you probably don't even need this tmp variable.

Reply:
named as `last_prob_dist`